saritasa_s3_tools/django/fields.py

```diff
@@ -0,0 +1,142 @@
+import typing
+import urllib.parse
+
+from django.conf import settings
+from django.core import validators
+from django.core.files.storage import default_storage
+from django.utils.translation import gettext_lazy as _
+from rest_framework import exceptions, fields, serializers
+
+import botocore.exceptions
+
+from .. import configs
+
+
+class S3FileTypeConfig(serializers.ChoiceField):
+    """Custom Choice field for s3 configs."""
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(choices=(), **kwargs)
+
+    def _get_choices(self) -> dict[str, str]:
+        """Get choices from S3FileTypeConfig."""
+        current_choices = tuple(
+            (
+                config_name,
+                config_name,
+            )
+            for config_name in configs.S3FileTypeConfig.configs
+        )
+        choices = super()._get_choices()
+        if current_choices != choices:
+            self._set_choices(current_choices)
+        return super()._get_choices()
+
+    def _set_choices(self, choices: tuple[tuple[str, str], ...]) -> None:
+        """Update choices.
+
+        Redefined to avoid recursion.
+
+        """
+        self.grouped_choices = fields.to_choices_dict(choices)
+        self._choices = fields.flatten_choices_dict(
+            self.grouped_choices,
+        )
+
+        # Map the string representation of choices to the underlying value.
+        # Allows us to deal with eg. integer choices while supporting either
+        # integer or string input, but still get the correct datatype out.
+        self.choice_strings_to_values = {
+            str(key): key for key in self._choices
+        }
+
+    choices = property(_get_choices, _set_choices)
+
+    def to_internal_value(
+        self,
+        data: typing.Any,
+    ) -> configs.S3FileTypeConfig | None:
+        """Convert api data to S3FileTypeConfig."""
+        try:
+            return configs.S3FileTypeConfig.configs[str(data)]
+        except KeyError:
+            self.fail("invalid_choice", input=data)
+
+
+class S3UploadURLField(serializers.URLField):
+    """URL serializer field for S3 `files or keys or URLs or objects`.
+
+    Convert url to a valid s3 key for bucket. On validation check that
+    key is present in bucket.
+
+    Example:
+    -------
+    https://www.saritasa-s3-tools.s3.minio.localhost/some-folder/file.txt will
```

Comment:
Let's improve example: add `aws-location` to URL and describe that AWS_LOCATION will be removed and only `key` will be returned